lib/models/BucketInfo.ts

```diff
@@ -38,44 +39,54 @@ export type VersioningConfiguration = {
     MfaDelete: any;
 };
 
-export type VeeamSOSApi = {
-    SystemInfo?: {
-        ProtocolVersion: string,
-        ModelName: string,
-        ProtocolCapabilities: {
-            CapacityInfo: boolean,
-            UploadSessions: boolean,
-            IAMSTS?: boolean,
-        },
-        APIEndpoints?: {
-            IAMEndpoint: string,
-            STSEndpoint: string,
-        },
-        SystemRecommendations?: {
-            S3ConcurrentTaskLimit: number,
-            S3MultiObjectDelete: number,
-            StorageCurrentTasksLimit: number,
-            KbBlockSize: number,
-        }
-        LastModified?: string,
-    },
-    CapacityInfo?: {
-        Capacity: number,
-        Available: number,
-        Used: number,
-        LastModified?: string,
-    },
-};
-
-// Capabilities contains all specifics from external products supported by
-// our S3 implementation, at bucket level
+/**
+ * Capabilities is the schema for the Capabilities object, where the
+ * capacity-related fields are bigints. Used by nodejs internally.
+ */
 export type Capabilities = {
-    VeeamSOSApi?: VeeamSOSApi,
+    VeeamSOSApi?: VeeamSOSApiSchema,
 };
 
 export type ACL = OACL & { WRITE: string[] }
 
-export default class BucketInfo {
+export type BucketMetadata = {
+    acl: ACL;
+    name: string,
+    owner: string,
+    ownerDisplayName: string,
+    creationDate: string,
+    mdBucketModelVersion: number,
+    transient: boolean,
+    deleted: boolean,
+    serverSideEncryption?: SSE,
+    versioningConfiguration?: VersioningConfiguration,
+    locationConstraint?: string,
+    readLocationConstraint?: string,
+    websiteConfiguration?: WebsiteConfigurationParams,
+    cors?: CORS,
+    replicationConfiguration?: any,
+    lifecycleConfiguration?: any,
+    bucketPolicy?: any,
+    uid: string,
+    isNFS?: boolean,
+    ingestion?: { status: 'enabled' | 'disabled' },
+    azureInfo?: any,
+    objectLockEnabled?: boolean,
+    objectLockConfiguration?: any,
+    notificationConfiguration?: any,
+    tags: Array<BucketTag>,
+    capabilities?: Capabilities,
+    quotaMax: bigint | number,
+};
+
+export type BucketMetadataJSON = Omit<Omit<BucketMetadata, 'quotaMax'>, 'capabilities'> & {
+    quotaMax: string;
+    capabilities: {
+        VeeamSOSApi?: VeeamSOSApiSchema,
```

Comment:
is this not the same as the "original" type?

Reply:
Updated and adapted the logic post type change